### DISCUSSION TITLE
How to perform Subset Similarity Search

### DISCUSSION THREAD

**Discussion by bigohk** [facebookresearch/faiss#3167] [Q&A]
Hi,

I am trying to perform a search.

The initial data is in text form. An example of the data is
```
name,interests
John Philip,Soccer Basketball Hockey
Sarah Smith Chase,Hockey Lacrosse Basketball
```

I generate embeddings for the above using `SentenceTransformer('bert-base-nli-mean-tokens')`

Then I instantiate a `faiss.IndexFlatL2`  and try a run the following queries

```python
queries = [
  'John Philip, Basketball Hockey Soccer',
  'John Philip, Basketball Soccer Hockey',
  'John Philip, Hockey Basketball Soccer',
  'John Philip, Hockey Soccer Basketball',
  'John Philip, Soccer Hockey Basketball',
  'John Philip, Soccer Basketball',
  'John Philip, Soccer Hockey',
  'John Philip, Basketball Hockey',
  'John Philip, Basketball',
  'John Philip, Hockey',
  'John Philip, Soccer',
]
```

The permutations of the full `interests` set result in the correct `John Philip` being returned.

However, the subsets do not.

What is the recommended way to perform this search ? Should I first produce the embeddings for subsets as well ?

Thanks

**Comment by heemin32** [comment]
I guess, you need embeddings for subsets. 
Also, https://github.com/facebookresearch/faiss/issues/3087 might help in your case if you want only single result for each group; For example, one for John Philip and another for Sarah Smith.

**Reply by bigohk** [reply]
Sorry, didn't get the chance to attend to this.

However, switching to a more "exhaustive" model has improved the results a lot.

So, the best performing ones are.. well... the obvious ones
* all-mpnet-base-v2
* all-MiniLM-L6-v2

Will report back if any other notable observation is made.

Thanks,
